ktor-server/ktor-server-netty/jvm/src/io/ktor/server/netty/NettyApplicationCallHandler.kt

```diff
@@ -27,8 +32,17 @@ internal class NettyApplicationCallHandler(userCoroutineContext: CoroutineContex
     }
 
     private fun handleRequest(context: ChannelHandlerContext, call: ApplicationCall) {
-        launch(CallHandlerCoroutineName + NettyDispatcher.CurrentContext(context), start = CoroutineStart.UNDISPATCHED) {
-            enginePipeline.execute(call)
+        val callContext = CallHandlerCoroutineName + NettyDispatcher.CurrentContext(context)
+
+        launch(callContext, start = CoroutineStart.UNDISPATCHED) {
+            when {
+                call is NettyHttp1ApplicationCall && call.request.httpRequest.decoderResult().isFailure -> {
```

Comment:
Is there a test for this particular case?

Reply:
Yep, the test I added